python/pyspark/sql/tests.py

```diff
@@ -3493,19 +3495,42 @@ def create_pandas_data_frame(self):
         data_dict["4_float_t"] = np.float32(data_dict["4_float_t"])
         return pd.DataFrame(data=data_dict)
 
-    def test_unsupported_datatype(self):
-        schema = StructType([StructField("map", MapType(StringType(), IntegerType()), True)])
-        df = self.spark.createDataFrame([(None,)], schema=schema)
-        with QuietTest(self.sc):
-            with self.assertRaisesRegexp(Exception, 'Unsupported type'):
-                df.toPandas()
+    @contextmanager
+    def arrow_fallback(self, enabled):
```

Comment:
I think it would be best to disable fallback for all the tests on setup/teardown.  That way if something goes wrong elsewhere, the tests won't start passing due to falling back. For the test where it is enabled, you could do that explicitly. What do you think?

Reply:
Yup, makes sense. Will give a shot.

BTW, while we are here, I was thinking of adding a more generalized version of an util like `arrow_fallback` to reduce configuration specific codes in the test scope but was hesitant because this approach is new to PySpark. WDTY? I will do another PR for this cleanup if we all feel in the same way. Cc @ueshin too.